app/controllers/hyrax/fixity_checks_controller.rb

```diff
@@ -2,15 +2,27 @@ module Hyrax
   class FixityChecksController < ApplicationController
     before_action :authenticate_user!
 
+    # request here with param :file_set_id will trigger a fixity check if
+    # needed, and respond with a JSON hash that looks something like:
+    #
+    #     { "file_id" => [
+    #         {
+    #           "checked_uri" => "http://127.0.0.1:8986/rest/test/12/57/9s/28/12579s28n/files/3ff48171-f625-48bb-a73d-b1ba16dde530/fcr:versions/version1",
+    #           "passed" => true,
+    #           "expected_result" => "urn:sha1:03434..."
+    #           "created_at" => "2017-05-16T15:32:50.961Z"
+    #         }
+    #       ]
+    #     }
     def create
       render json: fixity_check_service.fixity_check
     end
 
     protected
 
       def fixity_check_service
-        file_set = ::FileSet.find(params[:file_set_id])
-        FileSetFixityCheckService.new(file_set)
+        @fixity_check_service ||=
```

Comment:
I can add docs about async:false. We can also discuss whether the design is wise. 

Basically, because async:false is the only way to be assured to get a response that you can return immediately. The alternative would be more complicated code that somehow returned "in progress" for bytestreams without a fresh fixity check record. 

I went with the simple approach in part because I don't believe this action is actually _used by anyone_.  It's not used by existing hyrax code so far as I can tell. I think it's a leftover from a previous (pre FC4 even?) version of sufia that used it for "fixity status" display, but it hasn't for a while. 

I wanted to leave the action here because it actually _does_ seem to useful to have an HTTP API on fixity status for external integration.  But I didn't want to spend _too_ much time on it making it _perfect_, when just guessing on use cases. 

Another alternative would be removing this action/controller altogether, which I'd also be open to.